File: backend/omegatechapi/src/main/java/com/example/omegatechapi/model/TecnicoProfile.java
package com.example.omegatechapi.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@Entity
@Table(name = "tecnicos_perfis") // Nome da tabela no banco de dados, substituir depois

public class TecnicoProfile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String matricula;

    @Column(nullable = false)
    private String especialidade;

    private LocalDate dataCertificacao;

    @OneToOne
    @JoinColumn(name = "usuario_id", referencedColumnName = "id")
    private Usuario usuario;
}

File: client-desktop/model/TecnicoProfile.cs
using System.Text.Json.Serialization;

namespace model {
    public class TecnicoProfile {
        [JsonPropertyName("id")]
        public long Id { get; set; } 

        [JsonPropertyName("matricula")]
        public string Matricula { get; set; } 

        [JsonPropertyName("especialidade")]
        public string Especialidade { get; set; } 

        [JsonPropertyName("dataCertificacao")]
        public DateTime DataCertificacao { get; set; } 

        [JsonPropertyName("usuario")]
        public Usuario Usuario { get; set; }
    }
}